test/integration/BasketTest.php

```diff
@@ -241,6 +244,291 @@ public function chargeTransactionsShouldCreateBasketIfItDoesNotExistYet(): void
         $this->assertEquals($basket->expose(), $fetchedPayment->getBasket()->expose());
     }
 
+    //</editor-fold>
+
+    //<editor-fold desc="Basket v2 tests">
+
+    /**
+     * Verify basket can be created and fetched.
+     *
+     * @test
+     */
+    public function minV2BasketShouldBeCreatableAndFetchable(): void
+    {
+        $basket = new Basket();
+        $basket->setTotalValueGross(100)
+            ->setCurrencyCode('EUR');
+        $basketItem = new BasketItem();
+        $basketItem->setBasketItemReferenceId('item1')
+            ->setQuantity(1)
+            ->setAmountPerUnitGross(100)
+            ->setTitle('title');
+        $basket->addBasketItem($basketItem);
+        $this->assertEmpty($basket->getId());
+
+        $this->unzer->createBasket($basket);
+        $this->assertNotEmpty($basket->getId());
+
+        $invoice = new Invoice();
+        $invoice->setParentResource($this->unzer)->charge(100, 'EUR', 'https://unzer.com', null, 'orderId', null, $basket);
+        $fetchedBasket = $this->unzer->fetchBasket($basket->getId())->setOrderId('');
+        $this->assertEquals($basket->expose(), $fetchedBasket->expose());
+    }
+
+    /**
+     * Verify basket can be created and fetched.
+     *
+     * @test
+     */
+    public function V2BasketShouldBeCreatableAndFetchable(): void
+    {
+        $orderId = microtime(true);
+        $basket = new Basket();
+        $basket->setTotalValueGross(100)
+            ->setOrderId('testOrderId')
+            ->setCurrencyCode('EUR');
+        $basketItem = new BasketItem();
+        $basketItem->setBasketItemReferenceId('item1')
+            ->setQuantity(1)
+            ->setAmountPerUnitGross(100)
+            ->setTitle('title');
+        $basket->addBasketItem($basketItem);
+        $this->assertEmpty($basket->getId());
+
+        $this->unzer->createBasket($basket);
+        $this->assertNotEmpty($basket->getId());
+
+        $invoice = new Invoice();
+        $invoice->setParentResource($this->unzer)->charge(100, 'EUR', 'https://unzer.com', null, 'orderId', null, $basket);
+        $fetchedBasket = $this->unzer->fetchBasket($basket->getId());
+        $this->assertEquals($basket->expose(), $fetchedBasket->expose());
+    }
+
+    /**
+     * Verify basket can be created and fetched.
+     *
+     * @test
+     */
+    public function maxV2BasketShouldBeCreatableAndFetchableWorkAround(): void
```

Comment:
please describe the "workaround" in the comment of the function

Reply:
This test doesnt actually test the workaround. So i will remove it from the functions name. Of cause I will still add the comment to the "fetschBasket" funktion.